eclass/meson-multilib.eclass

```diff
@@ -0,0 +1,131 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: meson-multilib.eclass
+# @MAINTAINER:
+# gx86-multilib team <multilib@gentoo.org>
+# @AUTHOR:
+# Author: Michał Górny <mgorny@gentoo.org>
+# Author: Matt Turner <mattst88@gentoo.org>
+# @SUPPORTED_EAPIS: 7
+# @BLURB: meson wrapper for multilib builds
+# @DESCRIPTION:
+# The meson-multilib.eclass provides a glue between meson.eclass(5)
+# and multilib-minimal.eclass(5), aiming to provide a convenient way
+# to build packages using meson for multiple ABIs.
+#
+# Inheriting this eclass sets IUSE and exports default multilib_src_*()
+# sub-phases that call meson phase functions for each ABI enabled.
+# The multilib_src_*() functions can be defined in ebuild just like
+# in multilib-minimal, yet they ought to call appropriate meson
+# phase rather than 'default'.
+
+if [[ -z ${_MESON_MULTILIB_ECLASS} ]] ; then
+_MESON_MULTILIB_ECLASS=1
+
+case ${EAPI:-0} in
+	7) ;;
+	*) die "EAPI=${EAPI} is not supported" ;;
+esac
+
+inherit meson multilib-minimal
+
+EXPORT_FUNCTIONS src_configure src_compile src_test src_install
+
+# @FUNCTION: meson_native_bool
+# @USAGE: <USE flag> [option name]
+# @DESCRIPTION:
+# Given a USE flag and a meson project option, output a string like:
+#
+#   -Doption=true
+#   -Doption=false
+#
+# if building for the native ABI (multilib_is_native_abi is true). Otherwise,
+# output -Doption=false. If the project option is unspecified, it defaults
+# to the USE flag.
+meson_native_bool() {
```

Comment:
Personally, I find the naming to be confusing but I get your point. Still, I'd prefer if stuff using USE flags had `use` in its name.

Reply:
What should the names be then? How about:

`meson.eclass`:
* `meson_use` → `meson_use_bool`
* `meson_feature` → `meson_use_feature`

`meson-multilib.eclass`:
* `meson_native_bool` → `meson_native_use_bool`
* `meson_native_feature` → `meson_native_use_feature`
* `meson_native_true`
* `meson_native_enabled`

Seems to fit with `use_enable` and `use_with`.

Reply:
I would go a step further and use either `bool` or `true` consistently, e.g. `meson_native_use_bool` and `meson_native_bool` etc.

Reply:
Discussed on IRC and mgorny said

> <mgorny> mattst88: at a second look, i'd probably use _true and _enabled

sam and floppym acked.